src/components/checkbox-group/bl-checkbox-group.ts

```diff
@@ -0,0 +1,135 @@
+import { FormControlMixin } from '@open-wc/form-control';
+import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import 'element-internals-polyfill';
+import { event, EventDispatcher } from '../../utilities/event';
+import style from './bl-checkbox-group.css';
+import BlCheckbox, { blCheckboxTag } from './checkbox/bl-checkbox';
+
+export const blCheckboxGroupTag = 'bl-checkbox-group';
+
+export const blChangeEventName = 'bl-checkbox-group-change';
+
+/**
+ * @tag bl-checkbox-group
+ * @summary Baklava Button component
+ *
+ * @cssproperty --bl-checkbox-direction - Can be used for showing checkbox options as columns instead of rows. Options are `row` or `column`
+ */
+@customElement(blCheckboxGroupTag)
+export default class BlCheckboxGroup extends FormControlMixin(LitElement) {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  /**
+   * Sets the checkbox group label
+   */
+  @property({ type: String })
+  label: string;
+
+  /**
+   * Set and gets the actual value of the field
+   */
+  @property({ type: Array, reflect: true })
+  value: string[] = [];
+
+  /**
+   * Sets option as required
+   */
+  @property({ type: Boolean, reflect: true })
+  required = false;
+
+  get options(): BlCheckbox[] {
+    return [].slice.call(this.querySelectorAll(blCheckboxTag));
+  }
```

Comment:
You may prefer to use `@queryAll(blCheckboxTag)` of lit.

Reply:
I tried and it didn't select checkboxes somehow. Could @queryAll selector not support selecting the elements in shadowRoot ?

Reply:
Main purpose of it is querying the elements inside the component shadowRoot. It basically does what you do here:

> A property decorator that converts a class property into a getter that executes a querySelectorAll on the element's renderRoot.
> https://lit.dev/docs/api/decorators/#queryAll

But it doesn't return an array of `BlCheckbox` It returns `NodeListOf<BlCheckbox>`. Maybe that could be the issue?

Reply:
```
 @queryAll(blCheckboxTag) checkboxes: NodeListOf<BlCheckbox>;

  get options(): BlCheckbox[] {
    console.log("checkboxes",this.checkboxes);
    return [].slice.call(this.checkboxes);
  }
```

![image](https://user-images.githubusercontent.com/70342517/205037250-8562d2de-b425-4e1f-b648-860263aa459a.png)

Is there something missing ?

Reply:
Could you give it a shot on your local @muratcorlu  ? I couldn't debug the problem and make it work properly.

Reply:
`@query` and `@queryAll` are filled after `firstUpdated` hook. When do you call this getter? Can it be before `firstUpdated`?

Reply:
Yes, I call it right after mount. I need it to check all the available or checked checkboxes. Just like we did in radio-group actually.